### PR TITLE
Query strings now cooperate with unix sockets

### LINKED ISSUE #1766: Using options.qs with a unix socket fails
When the following request is done:
```js
request({
    uri: 'http://unix:/var/run/docker.sock:/containers/json'
})
```
[Some juggling is done](https://github.com/request/request/blob/master/request.js#L258) on `request.uri` to make the unix socket work. `request.uri.host` becomes `/var/run/docker.sock`.

However, when a query string is provided by the user, for example: 
```js
request({
    uri: 'http://unix:/var/run/docker.sock:/containers/json',
    qs: { all: 'true' }
})
```
Parsing the query string [clobbers the unix socket](https://github.com/request/request/blob/master/request.js#L1162). `request.uri.host` reverts to `unix`.

### PR DESCRIPTION
**Pull request by JoshWillik**
Fixes #1766

### CODE REVIEW

File: tests/test-unix.js [hunk 1]
```diff
@@ -27,7 +31,12 @@ tape('setup', function(t) {
 })
 
 tape('unix socket connection', function(t) {
```
Comment (simov) [inline tests/test-unix.js]: I would like to have two separate tests, the original one without querystring and a new one with querystring. That way if someone removes the unix bits from the init function the tests will throw an error.

File: request.js [hunk 2]
```diff
@@ -1411,6 +1405,19 @@ Request.prototype.destroy = function () {
     self.response.destroy()
   }
 }
+Request.prototype.enableUnixSocket = function () {
```
Comment (simov) [inline request.js]: Not the best place to put this function. Notice the comment above - *Streams API*

File: request.js [hunk 3]
```diff
@@ -1163,6 +1153,10 @@ Request.prototype.qs = function (q, clobber) {
   self.url = self.uri
   self.path = self.uri.path
 
+  if ( self.uri.host === 'unix' ) {
```
Comment (simov) [inline request.js]: Minor style issue - remove spacing.

**Comment by simov** [comment]
Hey, @JoshWillik nice work on this one. I added a few comments.

File: request.js (hunk 2 above)
Comment (simov) [inline request.js]: Also not sure if that needs to be defined on the prototype as a public method, probably a function that accepts an `uri` object will be just fine.

Comment (JoshWillik) [inline request.js]: @simov Line 1249 sets `this.socketPath`, which requires access to `this`, no?

Comment (JoshWillik) [inline request.js]: I've fixed all the things you highlighted, but I can make it a seperate by doing `fn (uri, context){}` or something if you prefer.

**Comment by simov** [comment]
:+1:

**Comment by JoshWillik** [comment]
:D Thanks @simov

**Comment by JoshWillik** [comment]
@simov How often does request update the NPM package? I'd like to be able to stop doing my own query string assembly in my code :stuck_out_tongue:

**Comment by simov** [comment]
Usually when someone asks about it, or when there is a bugfix that affects a lot of people. Strictly non semver :) but that'll change at some point. I'll let you know when the new release is published on NPM.

**Comment by simov** [comment]
@JoshWillik 2.62 is published.

**Comment by JoshWillik** [comment]
Hooray! Thank you :D

**Pull request merged**